extensions/api/data-management/policydefinition/src/test/java/org/eclipse/dataspaceconnector/api/datamanagement/policy/service/PolicyServiceImplTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022 ZF Friedrichshafen AG
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *    ZF Friedrichshafen AG - Initial API and Implementation
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.policy.service;
+
+import org.eclipse.dataspaceconnector.policy.model.Policy;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.contract.offer.store.ContractDefinitionStore;
+import org.eclipse.dataspaceconnector.spi.policy.store.PolicyStore;
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.transaction.NoopTransactionContext;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.dataspaceconnector.api.result.ServiceFailure.Reason.CONFLICT;
+import static org.eclipse.dataspaceconnector.api.result.ServiceFailure.Reason.NOT_FOUND;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class PolicyServiceImplTest {
+
+    private final PolicyStore policyStore = mock(PolicyStore.class);
+    private final ContractDefinitionStore contractDefinitionStore = mock(ContractDefinitionStore.class);
+    private final TransactionContext dummyTransactionContext = new NoopTransactionContext();
+
+    private final PolicyServiceImpl policyService = new PolicyServiceImpl(dummyTransactionContext, policyStore, contractDefinitionStore);
+
+    @Test
+    void findById_shouldRelyOnPolicyStore() {
+        when(policyStore.findById("policyId")).thenReturn(createPolicy("policyId"));
+
+        var policy = policyService.findById("policyId");
+        var uidTest = policy.getUid();
+
+        String assetId = "policyId";
+        assertThat(policy).isNotNull().matches(hasId(assetId));
+    }
+
+    @Test
+    void query_shouldRelyOnPolicyStore() {
+        var policy = createPolicy("policyId");
+        when(policyStore.findAll(any(QuerySpec.class))).thenReturn(Stream.of(policy));
+        var policies = policyService.query(QuerySpec.none());
+
+        assertThat(policies).hasSize(1).first().matches(hasId("policyId"));
```

Comment:
as above, simplify to `assertThat(policies).containsExactly(policy);` which is also deeper testing (tests more than just the id field)